### ISSUE TITLE
options-resolver fails when ext-uopz is enabled

### ISSUE THREAD

**Issue by Niko9911** [symfony/symfony#29842]
**Symfony version(s) affected**: "symfony/options-resolver": "^3.0 || ^4.0"

**Description**  
```
PHP Fatal error:  Class 'array' not found in vendor/symfony/options-resolver/OptionsResolver.php on line 1004
PHP Fatal error:  Class 'array' not found in vendor/symfony/options-resolver/OptionsResolver.php on line 1004
PHP Stack trace:
PHP   1. {main}() vendor/friendsofphp/php-cs-fixer/php-cs-fixer:0
PHP   2. PhpCsFixer\Console\Application->__construct() vendor/friendsofphp/php-cs-fixer/php-cs-fixer:74
PHP   3. PhpCsFixer\Console\Command\DescribeCommand->__construct() /vendor/friendsofphp/php-cs-fixer/src/Console/Application.php:56
PHP   4. PhpCsFixer\FixerFactory->registerBuiltInFixers() vendor/friendsofphp/php-cs-fixer/src/Console/Command/DescribeCommand.php:77
PHP   5. PhpCsFixer\Fixer\Import\OrderedImportsFixer->__construct() vendor/friendsofphp/php-cs-fixer/src/FixerFactory.php:107
PHP   6. PhpCsFixer\Fixer\Import\OrderedImportsFixer->configure() vendor/friendsofphp/php-cs-fixer/src/AbstractFixer.php:57
PHP   7. PhpCsFixer\FixerConfiguration\FixerConfigurationResolver->resolve() vendor/friendsofphp/php-cs-fixer/src/AbstractFixer.php:156
PHP   8. Symfony\Component\OptionsResolver\OptionsResolver->resolve() vendor/friendsofphp/php-cs-fixer/src/FixerConfiguration/FixerConfigurationResolver.php:106
PHP   9. Symfony\Component\OptionsResolver\OptionsResolver->offsetGet() vendor/symfony/options-resolver/OptionsResolver.php:772
PHP  10. Symfony\Component\OptionsResolver\OptionsResolver->verifyTypes() vendor/symfony/options-resolver/OptionsResolver.php:877
```

**How to reproduce**  
Just run CS Fixer while `ext-uopz` is enabled.

**Possible Solution**  
Not known.

**Additional context**  
PHP 71,72,73 affected when uopz version 5.x.x is used.

Even if issue is happening in CS fixer the main issue is in symfony library.

**Comment by stof** [comment]
Please report that to the ext-uopz maintainers instead.

**Comment by xabbuh** [comment]
closing as explained

**Comment by Niko9911** [comment]
@xabbuh if you're sure it's uopz issue, why I can only reproduce this with Symfony :upside_down_face:

**Comment by xabbuh** [comment]
I don't know, because I have no idea how that extension works. But looking at the reported line it seems that `(\function_exists($func = 'is_'.$type) && $func($value))` fails with the extension enabled. So checking why `is_array()` is not recognised may be a hint to fix the root cause

**Comment by Niko9911** [comment]
FYI
https://github.com/krakjoe/uopz/issues/101

**Comment by krakjoe** [comment]
Can someone eli5 how to produce the fault please ? 

Preferably without docker because it won't have my compiler/tools ...

**Comment by stof** [comment]
if the analysis done by @xabbuh is right, the following may exhibit the error (but I don't have ext-uopz to confirm):

```php
<?php

$a = null;
$type = 'array';

if ($a instanceof $type) {
    echo 'really weird';
}
```

@Niko9911 can you check whether this indeed reproduces the issue ?

**Comment by Niko9911** [comment]
I'm out of office. Don't want install stuff on Windows 😖. Sorry 11.1.2019 19.41 Christophe Coevoet <notifications@github.com> kirjoitti:if the analysis done by @xabbuh is right, the following may exhibit the error (but I don't have ext-uopz to confirm):
<?php

$a = null;
$type = 'array';

if ($a instanceof $type) {
    echo 'really weird';
}
@Niko9911 can you check whether this indeed reproduces the issue ?

**Comment by yceruto** [comment]
I've compiled the latest https://pecl.php.net/get/uopz-5.1.0.tgz on Ubuntu for PHP 7.3 and no errors testing @stof's code snippet.

**Comment by krakjoe** [comment]
I too can't reproduce, and can't see the patch in zend vm that could produce anything, for a non-object there is no branch that takes any action, not even a cache read ... https://php-lxr.adamharvey.name/source/xref/PHP-7.3/Zend/zend_vm_def.h#6781

**Comment by Niko9911** [comment]
@krakjoe @yceruto @stof 

Please follow steps to reproduce.

1) Clone `git clone https://gitlab.com/niko9911/php-test-generator.git`
2) Install Dependencies `composer install`
3) Make sure Uopz is present. `"echo extension_loaded('uopz') ? 'Uopz Enabled!' : 'Please install Uopz';"`
4) Run CS Fixer `composer cs` or `vendor/bin/php-cs-fixer fix -vvvv`

With these steps I have been able to reproduce problem on multiple platforms. I've used mainly PHP 7.2  to reproduce this issue.

------------------------

**Quick way**
If you want same env as I used to reproduce this, please do following:
1) Run command`docker run -it niko9911/symfony-issue-29842:xx` (xx = 71/72/73)
2) Run CS Fixer `composer cs` or `vendor/bin/php-cs-fixer fix -vvvv`

_There is images available for 71,72,73, just change tag in docker image._

**Comment by Niko9911** [comment]
@krakjoe any news...?

**Comment by Niko9911** [comment]
Thanks. I just wanted to make sure you got these instructions. :+1:

**Comment by Niko9911** [comment]
This has been now fixed with UOPZ 6.0
